### PR TITLE
Adding initial vendor dropdown functionality to all pages.

### PR DESCRIPTION
**Pull request by frostyclawz**
Adding the initial vendor dropdown to all pages. These dropdowns allow for searching for specific configurations to be narrowed down and refined for ease of use when interacting with the application's core functionalities.

### CODE REVIEW

**Comment by frostyclawz** [comment]
Screenshot 2020-07-22 at 4 40 51 PM - Display 2
Screenshot 2020-07-22 at 4 40 59 PM - Display 2
Screenshot 2020-07-22 at 4 41 04 PM - Display 2
Screenshot 2020-07-22 at 4 41 10 PM - Display 2

File: src/main/webapp/scripts/appscript.js [hunk 1]
```diff
@@ -17,25 +17,53 @@
...
-  console.log(customerJson);
+  var vendorJson = await response.json();
 
   // Loops over this array to make the options for the Customer ID dropdown.
   var vendorHtml = '';
-  for (var i = 0; i < customerJson.length; i++) {
-    html += '<option value="' + customerJson[i] + '">'
-    + customerJson[i] + '</option>';
+  for (var key in vendorJson) {
+    console.log(key);
```
Comment (cxfloeder) [inline src/main/webapp/scripts/appscript.js:28]: You could delete this console.log() line.

File: src/main/webapp/scripts/appscript.js [hunk 2]
```diff
@@ -17,25 +17,53 @@
...
+  var accountHtml = '';
+  for (var key in vendorJson) {
+    if (key == vendorValue) {
+      var array = vendorJson[key];
+      for (i = 0; i < array.length; i++) {
+        accountHtml += '<option value="' + array[i] + '">'
+        + array[i] + '</option>';
+      };
+    };
+  };
```
Comment (cxfloeder) [inline src/main/webapp/scripts/appscript.js:61]: Why are there semicolons here?

File: src/main/webapp/scripts/appscript.js [hunk 3]
```diff
@@ -17,25 +17,53 @@
...
 
   // Loops over this array to make the options for the Customer ID dropdown.
   var vendorHtml = '';
-  for (var i = 0; i < customerJson.length; i++) {
-    html += '<option value="' + customerJson[i] + '">'
-    + customerJson[i] + '</option>';
+  for (var key in vendorJson) {
+    console.log(key);
+    vendorHtml += '<option value="' + key + '">'
+    + key + '</option>';
```
Comment (cadecrandall) [inline src/main/webapp/scripts/appscript.js:30]: Constructing strings in JS could be a little bit cleaner if you used something like [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)

File: src/main/webapp/scripts/appscript.js [hunk 4]
```diff
@@ -17,25 +17,52 @@
...
+
+  var vendorJson = await response.json();
+
+  // Takes the value of the Vendor ID <select> element.
+  var vendorValue = document.getElementById('customer-ids').value;
+
+  // Loops over the dictionary to find the corresponding value in the dictionary
+  // and adds them to the Account ID <select> element.
+  var accountHtml = '';
+  for (var key in vendorJson) {
```
Comment (ianjjohnson) [inline src/main/webapp/scripts/appscript.js:52]: no need to loop through the keys here. You can remove the loop and just say
`var array = vendorJson[vendorValue]`.

Consider renaming "array" to indicate what it actually holds

**Pull request merged**